spec/strava/api/client/endpoints/uploads/create_upload_spec.rb

```diff
@@ -10,13 +10,35 @@
       file: Faraday::UploadIO.new(file, 'application/tcx+xml'),
       name: 'Uploaded Activity',
       description: 'Uploaded by strava-ruby-client.',
-      data_type: 'tcx',
-      external_id: 'strava-ruby-client-upload-1'
+      data_type: 'tcx'
     )
-    expect(upload.id).to eq 2_136_460_097
-    expect(upload.external_id).to eq 'strava-ruby-client-upload-1.tcx'
+
+    expect(upload.id).to eq 8_908_667_008
     expect(upload.error).to eq nil
     expect(upload.status).to eq 'Your activity is still being processed.'
```

Comment:
this is exactly how I understood the process. when I query `client.upload(id-of-activity)` this will be the status until it's successful. 

I can capture the happy path's end result after 30 seconds of sleep, too. I will add this as an open todo.

Reply:
I'm just saying that semantically it's cleaner to check whether `upload.finished?` instead of `upload.id.nil?`. The implementation of `finished?` would just be `! id.nil?`.

Reply:
totally agree, thank you! But I'd much prefer `processing?` and `processed?` as it reflects Strava's terminology. Wdyt? Added `successful?` as a method.  
But I will totally bow to your final decision.

I extracted the upload status logic into a new mixin, to keep things tidy.

our cassette deck should reflect all cases and states now 🚀

I am not too fond of checking against a string, but with a chore/re-recording we could provide a sane amount of stability, I guess.
https://relishapp.com/vcr/vcr/v/6-1-0/docs/cassettes/automatic-re-recording

Reply:
I think the Strava one is better to match closer, yes to `processing?` and `processed?`.

Reply:
utilizing `processing?` to loop in the specs and get rid of the sleep hack